src/highlevel/forward_mode.jl

```diff
@@ -40,21 +45,23 @@ function sparse_jacobian_cache(ad::Union{AutoSparseForwardDiff, AutoForwardDiff}
 end
 
 function sparse_jacobian!(J::AbstractMatrix, _, cache::ForwardDiffJacobianCache, f::F,
-    x) where {F}
+        x) where {F}
     if cache.cache isa ForwardColorJacCache
         forwarddiff_color_jacobian(J, f, x, cache.cache) # Use Sparse ForwardDiff
     else
-        ForwardDiff.jacobian!(J, f, x, cache.cache) # Don't try to exploit sparsity
+        # Disable tag checking since we set the tag to our custom tag
+        ForwardDiff.jacobian!(J, f, x, cache.cache, Val(false)) # Don't try to exploit sparsity
```

Comment:
this is a nono

Reply:
But forwarddiff will check that the cache tag must match `Tag(f, v)`? How to deal with that